File: 179. Largest Number/solution.java
class Solution {
    public String largestNumber(int[] nums) {

        int n = nums.length;
        String[] store = new String[n];

        for (int i = 0; i < n; i++) {
            store[i] = String.valueOf(nums[i]);
        }
        Arrays.sort(store, (a, b) -> (b + a).compareTo(a + b));

        if (store[0].equals("0")) {
            return "0";
        }

        StringBuilder ans = new StringBuilder();
        
        for (String s : store) {
            ans.append(s);
        }

        return ans.toString();
    }
}

File: 179. Largest Number/solution.cpp
class Solution {     
public:
    static bool comp(string a,string b){
        return a+b>b+a;
    }
       
     string largestNumber(vector<int>& nums) {
        
        int n  = nums.size();
        vector<string> store(n);
        
        for(int i=0;i<n;i++)
            store[i] = to_string(nums[i]);
        
        
        sort(store.begin(),store.end(),comp);
        string ans ="";
        
        for(auto x:store)
           ans+=x;
        
        return ans[0] =='0'?"0":ans;
          
    }
};